visualization/app/codeCharta/state/metric.service.ts

```diff
@@ -64,6 +74,20 @@ export class MetricService implements FileStateServiceSubscriber, SettingsServic
 		return metric ? metric.maxValue : undefined
 	}
 
+	public getAttributeTypeByMetric(metricName: string, settings: Settings): AttributeType {
+		let attributeType: AttributeType = null
```

Comment:
Can we refactor these loops to something like the following?
```
return settings.fileSettings.attributeTypes.nodes
    .find(x => Object.keys(x)[0] === metricName)
```

Reply:
refactored to
```
public getAttributeTypeByMetric(metricName: string, settings: Settings): AttributeType {
		const attributeType = settings.fileSettings.attributeTypes.nodes.find(x => _.findKey(x) === metricName)

		if (attributeType) {
			return attributeType[metricName]
		}
		return null
	}
```